functions/src/venue.ts

```diff
@@ -1141,3 +1141,68 @@ export const deleteScreeningRoomVideo = functions.https.onCall(
       .delete();
   }
 );
+
+export const upsertChannel = functions.https.onCall(async (data, context) => {
+  assertValidAuth(context);
+
+  const { spaceId, channelIndex, channel } = data;
+
+  const space = await getSpaceById(spaceId);
+
+  await throwErrorIfNeitherWorldNorSpaceOwner({
+    spaceId,
+    worldId: space.worldId,
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    userId: context.auth.token.user_id,
+  });
+
+  const doc = await admin.firestore().collection("venues").doc(spaceId).get();
```

Comment:
You are already getting the space above `const space = await getSpaceById(spaceId);` do you need this code? It might increase the request complete time, especially on slow internet.